### ISSUE TITLE
Change Input/Output Audio feed

### ISSUE THREAD

**Issue by klementuel** [bigbluebutton/bigbluebutton#9090]
**Is your feature request related to a problem? Please describe.**
I want to change during präsentation between to Microphones. When I try that with the system tools in IOs or Chrome, it crash the microphone feed and nobody can hear me.

**Describe the solution you'd like**
I´d like a solutions like in jitsi.
or maybe a setup like this
https://developers.google.com/web/updates/2015/10/media-devices

**Comment by klementuel** [comment]
No I want do change the audio device after the echo test

**Comment by matiasilva** [comment]
Interesting that the echo test takes 20-30 seconds for you. The latest test.bigbluebutton.org takes ~7 for me. If you are changing audio input devices it does make sense to go through the echo test again, it's an entirely new device.

**Comment by KimBrodowski** [comment]
Not really if the devices is known and working. As stated in the original description you might just have multiple devices. I don't configure my SIP client every call either.

The echo test time issue is tracked in #8824.
tl;dr: It heavily depends on your browser and network setup. Up to 30 seconds isn't uncommon.

**Comment by ritzalam** [comment]
The echo test isn't just about testing the device but also includes testing the complete path of the audio. Testing the full path ensures that media packets are not blocked (closed UDP ports) and signaling is working.

We could probably split this into two: (1) device test and (2) audio test.

**Comment by KimBrodowski** [comment]
Without having implemented any of this in a browser myself I would expect you can just reuse the already existing connection? Establishing a new one could mean having to go through ICE again which is what takes so long in the first place.

**Comment by matiasilva** [comment]
Hmmm interesting..Am looking forward to seeing if that has any measurable improvements for @KimBrodowski

**Comment by prlanzarin** [comment]
Input device swapping can be done on the fly, no need to renegotiate anything (see `replaceTrack`). We just need to bolt it in when we have the time to do so because it's a bit tricky (not difficult, just tricky).

The aforementioned ICE settings only do wonders for Kurento related stuff (it **really** speeds things up for video, screensharing and listen only), but it does not relate to echo test at all so it won't have any effect on that.

**Comment by KimBrodowski** [comment]
I actually do that on our 2.3 alpha setup, but yeah no effect on mod_sofia. It just doesn't appear to be very deterministic aside from there clearly being 2 different time levels.

Attempts against test.bigbluebutton.org with FF 81.0 on Ubuntu 20.04 amd64:
1. 6 seconds
2. 38 seconds
3. 37 seconds
4. 7 seconds

This is a bit off-topic here of course. Just clarification why implementing this is a good idea. I'd assume my dual stack network setup with 9 interfaces probably doesn't help. ;)

**Comment by prlanzarin** [comment]
> I'd assume my dual stack network setup with 9 interfaces probably doesn't help. ;)
Probably not. The best bet in this scenario is tinkering around with FreeSWITCH candidate ACL to see if it privileges the right stuff first instead of bogus candidates. Tread with care, though, because that ACL is tricky to deal with.

**Comment by antobinary** [comment]
Is this the same issue as https://github.com/bigbluebutton/bigbluebutton/issues/9723 ?
